api/src/controllers/validation.ts

```diff
@@ -84,5 +83,23 @@ const beProperlyUriEncoded = {
     message: 'Must be properly encoded with encodeURIComponent',
 };
 
-export { beAResumeReviewState, beAValidUser, beAValidUuid, beProperlyUriEncoded };
+/**
+ * Test whether a field is a valid url.
+ */
+const beAValidUrl = {
+    predicate: async (field: string | undefined): Promise<boolean> => {
+        if (field === undefined) {
+            return false;
+        }
```

Comment:
Nit: we can collapse this to the try catch, `new URL(undefined)` fails
![image](https://user-images.githubusercontent.com/43416725/125718639-bdacd0c8-06a5-454b-ad29-35e43a3ab694.png)

Reply:
Good catch but going to keep it as is. Without the if statement the compiler complains about assigning `string?` to `string` and I would need to do something like `field!`

Reply:
Ah good call, sounds good!